### PR TITLE
Improve: inline extensions that do not break

### LINKED ISSUE #1048: Bug: top-level extension points are misindented based on what precedes them
**Describe the bug**
I get this formatting:
```
let this_function_has_a_long_name plus very many arguments = "and a kind of long body"

[%%expect
  {||}]
```
The `[%%expect {||}]` extension should be all on one line, not split onto two lines.

**How to Reproduce**
I got this using ocamlformat built from ce9e53933cf80fe164bc12dac31913d33b0f6e3d, using `--profile=janestreet`.

### PR DESCRIPTION
**Pull request by gpetiot**
Fix #1048 
We put a box around non-breaking payload of the extensions (the breaking ones must not be boxed)

I think in the future we should replace the calls to `Location.is_single_line` and their friends with `fits` whenever possible, so we rely on the output instead of the input.

No regression with test_branch.

### CODE REVIEW

File: src/Fmt_ast.ml [hunk 1]
```diff
@@ -562,14 +562,24 @@ let sequence_blank_line c xe1 xe2 =
...
 let rec fmt_extension c ctx key (ext, pld) =
   match (pld, ctx) with
   | ( PStr [({pstr_desc= Pstr_value _ | Pstr_type _; _} as si)]
     , (Pld _ | Str _ | Top) ) ->
       fmt_structure_item c ~last:true ~ext (sub_str ~ctx si)
   | PSig [({psig_desc= Psig_type _; _} as si)], (Pld _ | Sig _ | Top) ->
       fmt_signature_item c ~ext (sub_sig ~ctx si)
-  | _ -> fmt_attribute_or_extension c key Fn.id (ext, pld)
+  | _ ->
+      let fits = fits c pld ~f:(fmt_payload c (Pld pld)) in
```
Comment (Julow) [inline src/Fmt_ast.ml]: Why is this necessary ? Is this not possible to implement with `fits_breaks` ?

File: test/passing/doc_comments-after.ml.ref [hunk 2]
```diff
@@ -232,8 +232,7 @@ end = struct
   (* [@@@some attribute] *)
   (* (** Attribute *) *)
 
-  (** Extension *)[%%some
-  extension]
+  (** Extension *)[%%some extension]
```
Comment (Julow) [inline test/passing/doc_comments-after.ml.ref]: Not related but there should be a break after the doc comment.

File: src/Fmt_ast.ml (hunk 1 above)
Comment (gpetiot) [reply src/Fmt_ast.ml]: No because we would need to open the box before formatting (before using fits_breaks)

File: test/passing/doc_comments-after.ml.ref (hunk 2 above)
Comment (gpetiot) [reply test/passing/doc_comments-after.ml.ref]: It can indeed be fixed in another PR.
